### DISCUSSION TITLE
Searching an on-line zip file while streaming remotely

### DISCUSSION THREAD

**Discussion by IngenioCraft** [BurntSushi/ripgrep#2341] [Q&A]
Any ideas how to use ripgrep to search a zip file on a website without downloading the entire file.
I have a list of very large files and want to try to figure out which ones have relevant content without downloading them all locally.

For example, I am looking for a list of files that contain a specific string in one of the files listed here: 
https://transparency-in-coverage.uhc.com/?_gl=1*dlu06b*_ga*NzcyNDUyMi4xNjY0NjI4Nzg3*_ga_HZQWR2GYM4*MTY2NjgyODcxMS4yLjAuMTY2NjgyODcxMS4wLjAuMA..

**Comment by BurntSushi** [comment]
The only way to search a corpus without downloading it is for the thing that provides the corpus to provide an index. Otherwise I'm not sure what you're expecting here... In order for ripgrep to search something, it has to be made available through your file system and readable via standard `read` system calls.

**Reply by IngenioCraft** [reply]
Thanks so much for your quick reply!  Can you think of another tool that may accomplish this task?

**Reply by BurntSushi** [reply]
I don't mean to be rude, but consider reading the first sentence in my reply again.

If you want to search it and there is no service for doing so that someone else has already provided for you, then you have to download all of it. There's really not much else to say. Once you download it, there are lots of tools for searching it, including ripgrep. (Although exhaustively searching 1TB data is going to take quite some time.)
